File: src/main/java/org/intellij/sdk/language/angelscript/AngelScriptSyntaxHighlighterFactory.java
package org.intellij.sdk.language.angelscript;

import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public class AngelScriptSyntaxHighlighterFactory extends SyntaxHighlighterFactory {

    @Override
    public @NotNull SyntaxHighlighter getSyntaxHighlighter(Project project, VirtualFile virtualFile) {
        return new AngelScriptSyntaxHighlighter();
    }
}

File: src/main/kotlin/com/enderzombi102/css/lang/angelscript/AngelScriptSyntaxHighlighterFactory.kt
package com.enderzombi102.css.lang.angelscript

import com.intellij.openapi.fileTypes.SyntaxHighlighterFactory
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile

class AngelScriptSyntaxHighlighterFactory : SyntaxHighlighterFactory() {
	override fun getSyntaxHighlighter( project: Project?, virtualFile: VirtualFile? ) =
		AngelScriptSyntaxHighlighter()
}